TodoApp.ConsoleApp/Services/TodoService.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections.Generic;
+using TodoApp.Library.Data;
+using TodoApp.Library.Models;
+using System.Linq;
+using System;
+
+namespace TodoApp.ConsoleApp.Services
+{
+    public class TodoService : ITodoService
+    {
+        const string path = "../../tasks.xml";
+
+        public IEnumerable<Task> GetAll()
+        {
+            XMLTaskReader reader = new XMLTaskReader();
+
+            IEnumerable<Task> tasks = reader.ReadTasks();
+
+            return tasks;
+        }
+
+        public Task GetByID(string id)
+        {
+            IEnumerable<Task> tasks = GetAll();
+
+            Task wantedTask = null;
+
+            try
+            {
+                wantedTask = tasks.First(task => task.ID.ToString() == id);
+            }
+            catch (InvalidOperationException)
+            {
+                Console.WriteLine("Invalid id!");
+            }
+
+            return wantedTask;
+        }
+
+        public bool Update(Task task)
+        {
+            bool isUpdated = false;
+
+            Task updatedTask = null;
+            Create(updatedTask);
+
+            if (updatedTask != null)
+            {
+                Delete(task.ID.ToString());
+                isUpdated = true;
+            }
+
+            return isUpdated;
+        }
+
+        public Task Create(Task task)
+        {
+            XMLTaskWriter writer = new XMLTaskWriter();
+            writer.Save(task);
+
+            return task;
+        }
+
+        public bool Delete(string id)
+        {
+            bool isDeletable = false;
+            if (IsFoundId(id))
+            {
+                XMLTaskWriter writer = new XMLTaskWriter();
+                Task taskToBeDeleted = GetByID(id);
+                writer.Delete(taskToBeDeleted);
+                Console.WriteLine("Delete completed!");
+                isDeletable = true;
+            }
+            else
+            {
+                Console.WriteLine("Delete not complete!");
+            }
+
+            return isDeletable;
```

Comment:
You can just use `GetByID()`, and remove `IsFoundId()`:
```csharp
Task taskToBeDeleted = GetByID(id);
if (taskToBeDeleted == null)
{
  return false;
}

XMLTaskWriter writer = new XMLTaskWriter();
writer.Delete(taskToBeDeleted);
return true;
```

Don't use the `Console`.